src/helpers.ts

```diff
@@ -28,3 +33,165 @@ export async function pathExists(
   }
   return (stat.type & fileType) === fileType;
 }
+
+export class StopWatch {
+  private started = new Date().getTime();
+  public get elapsedTime() {
+      return new Date().getTime() - this.started;
+  }
+  public reset() {
+      this.started = new Date().getTime();
+  }
+}
+
+/**
+ * Accumulator to concatenate cell slices for a sliced program, preserving cell structures.
+ */
+export function concat(existingText: string, newText: ppa.CellSlice): string {
+  // Include our cell marker so that cell slices are preserved
+  return `${existingText}#%%\n${newText.textSliceLines}\n`;
+}
+
+/**
+ * This is called to convert VS Code ICells to Gather ICells for logging.
+ * @param cell A cell object conforming to the VS Code cell interface
+ */
+export function convertVscToGatherCell(cell: vscode.NotebookCell): ppa.Cell | undefined {
+  let code = '';
+  if (cell.document) {
+    code = cell.document.getText();
+  } else {
+    code = (<any>cell).code as string;
```

Comment:
This seems a little scary. There's an implicit dependency here between ICell (in the jupyter extension) and this code.

Reply:
Do we not translate ICells into vscode.NOtebookCells before sending them?

Reply:
There is, if the cell came from a native notebook, it'll have the document. If it was translated from an ICell, it'll have the code field.

Reply:
That seems bad. There's an implicit dependency here. Would it not be better to just make them use the same interface? (Maybe not NotebookCell, but rather something that just has the document URI and the code?)